soc/nxp/lpc/lpc54xxx/Kconfig.soc

```diff
@@ -1,47 +1,43 @@
 # LPC LPC54XXX MCU line
 
-# Copyright (c) 2017, NXP
+# Copyright (c) 2017, 2024 NXP
 # SPDX-License-Identifier: Apache-2.0
 
-choice
-	prompt "LPC LPC54XXX MCU Selection"
-	depends on SOC_SERIES_LPC54XXX
+config SOC_SERIES_LPC54XXX
+	bool
+	select SOC_FAMILY_LPC
+
+config SOC_SERIES
+	default "lpc54xxx" if SOC_SERIES_LPC54XXX
+
+config SOC_LPC54114
+	bool
+	select SOC_SERIES_LPC54XXX
 
 config SOC_LPC54114_M4
-	bool "SOC_LPC54114_M4"
-	select CPU_CORTEX_M4
-	select CPU_CORTEX_M_HAS_DWT
-	select CPU_HAS_ARM_MPU
-	select CPU_HAS_FPU
-	select PLATFORM_SPECIFIC_INIT
-	select CLOCK_CONTROL
-	select HAS_MCUX_IAP_LEGACY
+	bool
+	select SOC_LPC54114
 
 config SOC_LPC54114_M0
-	bool "SOC_LPC54114_M0"
-	select CPU_CORTEX_M0PLUS
-	select CPU_CORTEX_M_HAS_VTOR
-	select CLOCK_CONTROL
-
-endchoice
+	bool
+	select SOC_LPC54114
 
-if SOC_SERIES_LPC54XXX
+# The NXP HAL expects the SOC to just be lpc54114 if targeting
+# the M4
+config SOC
+	default "lpc54114" if SOC_LPC54114_M4
+	default "lpc54114_m0" if SOC_LPC54114_M0
```

Comment:
soc name is `lpc54114` for both

Reply:
So this breaks things for me because currently our HAL is using this and expecting lpc54114_m0 in CONFIG_SOC. I think we can flag this for addressing later as we (NXP) have been trying to avoid HAL changes to support hwmv2.

Reply:
@danieldegrasse is looking at doing a HAL_NXP PR to handle this type of issue as it is impacting the other platforms.